### ISSUE TITLE
HRA: cryptic error from mismatch between info & criteria tables

### ISSUE THREAD

**Issue by davemfish** [natcap/invest#654]
HRA validates (not in `validate`, but early in `execute`) that the criteria table has all the habitat and stressor names that appear in the info table. But it does not catch the reverse: when a habitat appears in the criteria table but is missing from the info table. HRA should raise a helpful `ValueError` early in runtime when this happens.

Right now, this bug results in unexpected behavior later in `_get_overlap_dataframe`, because that function ignores columns in the criteria table that come after any column headed by a name that is not included in the info table habitat list. For example:

```
Traceback (most recent call last):
  File "site-packages\natcap\invest\ui\model.py", line 1649, in _logged_target
  File "site-packages\natcap\invest\hra.py", line 273, in execute
  File "site-packages\natcap\invest\hra.py", line 2984, in _get_overlap_dataframe
ValueError: The following stressor-habitat pair(s) do not have at least one exposure rating: [('RCF', 'Roads'), ('RCF', 'MIN'), ('RCF', 'OPL'), ('RCF', 'SET'), ('RCF', 'DCL'), ('RCF', 'BRL'), ('UDM', 'Roads'), ('UDM', 'MIN'), ('UDM', 'OPL'), ('UDM', 'SET'), ('UDM', 'DCL'), ('UDM', 'BRL'), ('DIM', 'Roads'), ('DIM', 'MIN'), ('DIM', 'OPL'), ('DIM', 'SET'), ('DIM', 'DCL'), ('DIM', 'BRL'), ('WAB', 'Roads'), ('WAB', 'MIN'), ('WAB', 'OPL'), ('WAB', 'SET'), ('WAB', 'DCL'), ('WAB', 'BRL')]
```

**Comment by phargogh** [comment]
I'm working on this in my fork on `bugfix/654-cryptic-error-mismatch-betwixt-info-and-criteria-tables`.  I've got a couple minor other patches there in the model as well.

I've attached a sample patch of what I thought would be a nice, easy fix, but of course it's more complicated than that.  Need to more fully understand the structure of these inputs before I can validate them properly.
[sample_patch.txt](https://github.com/natcap/invest/files/7873641/sample_patch.txt)
